sentry-kotlin-multiplatform/src/appleMain/kotlin/io/sentry/kotlin/multiplatform/nsexception/SentryUnhandledExceptions.kt

```diff
@@ -0,0 +1,106 @@
+// https://github.com/rickclephas/NSExceptionKt/blob/master/nsexception-kt-sentry/src/commonMain/kotlin/com/rickclephas/kmp/nsexceptionkt/sentry/Sentry.kt
+//
+// Copyright (c) 2022 Rick Clephas
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+
+package io.sentry.kotlin.multiplatform.nsexception
+
+import NSException.Sentry.*
+import kotlinx.cinterop.UnsafeNumber
+import platform.Foundation.NSException
+import platform.Foundation.NSNumber
+
+/**
+ * Drops the Kotlin crash that follows an unhandled Kotlin exception.
+ */
+fun dropKotlinCrashEvent(event: SentryEvent?): SentryEvent? {
+    return event?.takeUnless { it.isCrashEvent && (it.tags?.containsKey(kotlinCrashedTag) ?: false) }
+}
+
+/**
+ * Sets the unhandled exception hook such that all unhandled exceptions are logged to Sentry as fatal exceptions.
+ * If an unhandled exception hook was already set, that hook will be invoked after the exception is logged.
+ * Note: once the exception is logged the program will be terminated.
+ * @see wrapUnhandledExceptionHook
+ */
+fun setSentryUnhandledExceptionHook(): Unit = wrapUnhandledExceptionHook { throwable ->
+    val envelope = throwable.asSentryEnvelope()
+    // The envelope will be persisted, so we can safely terminate afterwards.
+    // https://github.com/getsentry/sentry-cocoa/blob/678172142ac1d10f5ed7978f69d16ab03e801057/Sources/Sentry/SentryClient.m#L409
+    SentrySDK.storeEnvelope(envelope)
+    SentrySDK.configureScope { scope ->
+        scope!!.setTagValue(kotlinCrashedTag, kotlinCrashedTag)
+    }
```

Comment:
Would be possible to simply call `SentrySDK.captureEnvelope`?
Or even, instead of wrapping the event in an envelope, rather `SentrySDK.captureEvent`?

Reply:
I tried it and didn't work since this is being called right before the app terminates and `sendEnvelope` being in the background thread: https://github.com/getsentry/sentry-cocoa/blob/c51bd77615c8685569ad3da7bad7a001479ce110/Sources/Sentry/SentryHttpTransport.m#L92